integration/utils/try.go

```diff
@@ -1,50 +1,209 @@
+// processing at the call site.
+// Conditions are not allowed since it would complicate signaling if the
+// response body needs to be closed or not. Callers are expected to close on
+// their own if the function returns a nil error.
+func TryGetResponseUntilStatusCode(url string, timeout time.Duration, statusCode int) (*http.Response, error) {
+	return doGetResponse(url, timeout, UntilStatusCodeIs(statusCode))
+}
+
+// TryResponse is like TryRequest, but returns the response for further
+// processing at the call site.
+// Conditions are not allowed since it would complicate signaling if the
+// response body needs to be closed or not. Callers are expected to close on
+// their own if the function returns a nil error.
+func TryResponse(req *http.Request, timeout time.Duration) (*http.Response, error) {
+	return doResponse(req, timeout, nil)
+}
+
+// TryResponseUntilStatusCode is like TryRequest, but returns the response for further
+// processing at the call site.
+// Conditions are not allowed since it would complicate signaling if the
+// response body needs to be closed or not. Callers are expected to close on
+// their own if the function returns a nil error.
+func TryResponseUntilStatusCode(req *http.Request, timeout time.Duration, statusCode int) (*http.Response, error) {
+	return doResponse(req, timeout, UntilStatusCodeIs(statusCode))
+}
+
+// TryGetRequest is like Try, but runs a request against the given URL and applies
+// the condition on the response.
+// Condition may be nil, in which case only the request against the URL must
+// succeed.
+func TryGetRequest(url string, timeout time.Duration, condition Condition) error {
+	resp, err := doGetResponse(url, timeout, condition)
+
+	if resp != nil && resp.Body != nil {
+		defer resp.Body.Close()
+	}
+
 	return err
 }
 
-// Try try operation timeout, and retry backoff
-func Try(timeout time.Duration, operation func() error) error {
-	exponentialBackOff := backoff.NewExponentialBackOff()
-	exponentialBackOff.MaxElapsedTime = timeout
-	err := backoff.Retry(operation, exponentialBackOff)
+// TryRequest is like Try, but runs a request against the given URL and applies
+// the condition on the response.
+// Condition may be nil, in which case only the request against the URL must
+// succeed.
+func TryRequest(req *http.Request, timeout time.Duration, condition Condition) error {
+	resp, err := doResponse(req, timeout, condition)
+
+	if resp != nil && resp.Body != nil {
+		defer resp.Body.Close()
+	}
+
 	return err
 }
 
+// Try repeatedly executes an operation until no error condition occurs or the
+// given timeout is reached, whatever comes first.
+func Try(timeout time.Duration, operation func() error) error {
+	if timeout <= 0 {
+		panic("timeout must be larger than zero")
+	}
+
+	interval := time.Duration(math.Ceil(float64(timeout) / 15.0))
+	if interval > maxInterval {
+		interval = maxInterval
+	}
+
+	ci := os.Getenv("CI")
```

Comment:
> Currently, this doesn't work because we don't access to global env var in the integration test.

> hum.. :thinking: we have access to whichever the environment variable used to execute the tests, so we just need to pass them.

in our repo via .semaphoreci/vars we can pass global vars to the CI, but in this case the issue is the overlay docker level we add when we want execute the tests so the the CI global vars need to be passed in the make file like that: https://github.com/atbore-phx/traefik/commit/6aef257b5f00231854ab564305796ee8db78500a